apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget_Configuration.cs

```diff
@@ -0,0 +1,135 @@
+        /// </summary>
+        public Layout ProjectId { get; set; }
+
+        /// <summary>
+        /// LogID for all log entries. Defaults to "Default".
+        /// </summary>
+        public Layout LogId { get; set; } = "Default";
+
+        /// <summary>
+        /// Specify labels for the resource type;
+        /// only used if platform detection is disabled or detects an unknown platform.
+        /// </summary>
+        [ArrayParameter(typeof(TargetPropertyWithContext), "resourcelabel")]
+        public IList<TargetPropertyWithContext> ResourceLabels { get; }
+
+        /// <inheritdoc/>
+        [ArrayParameter(typeof(TargetPropertyWithContext), "contextproperty")]
+        public override IList<TargetPropertyWithContext> ContextProperties { get { return _contextProperties; } }
+        private readonly List<TargetPropertyWithContext> _contextProperties;
+
+        /// <summary>
+        /// How many seconds to wait for task completion before starting new task
+        /// </summary>
+        public int TimeoutSeconds { get; set; } = 2;
+
+        /// <summary>
+        /// Fills <see cref="LogEntry.JsonPayload"/> instead of <see cref="LogEntry.TextPayload"/>.
+        /// JSON serialization can be customed using <see cref="JsonConvertTypeName"/> and <see cref="JsonConvertMethodName"/>,
+        /// or <see cref="JsonConvertFunction"/>.
+        /// </summary>
+        public bool SendJsonPayload { get; set; }
+
+        /// <summary>
+        /// When <see cref="SendJsonPayload"/> is <c>true</c>, a custom JSON serialization method may be specified by
+        /// type-name and method-name. Specify the type-name here, and the method-name in <see cref="JsonConvertMethodName"/>.
+        /// The instance or static method must have the signature <c>Value MyMethod(object obj)</c> or <c>JToken MyMethod(object o)</c>.
+        /// Must not be set if <see cref="JsonConvertFunction"/> is set.
+        /// </summary>
+        public string JsonConvertTypeName { get; set; }
+
+        /// <summary>
+        /// When <see cref="SendJsonPayload"/> is <c>true</c>, a custom JSON serialization method may be specified by
+        /// type-name and method-name. Specify the type-name in <see cref="JsonConvertTypeName"/>, and the method-name here.
+        /// The instance or static method must have the signature <c>Value MyMethod(object obj)</c> or <c>JToken MyMethod(object o)</c>.
+        /// Must not be set if <see cref="JsonConvertFunction"/> is set.
+        /// </summary>
+        public string JsonConvertMethodName { get; set; }
+
+        /// <summary>
+        /// When <see cref="SendJsonPayload"/> is <c>true</c>, a custom JSON serialization method may be programatically specified here.
+        /// Must not be set if <see cref="JsonConvertTypeName"/> and <see cref="JsonConvertMethodName"/> are set.
+        /// </summary>
+        public Func<object, Value> JsonConvertFunction { get; set; }
```

Comment:
How about calling this `JsonConverter`, but the others `JsonConverterTypeName` and `JsonConverterMethodName`?

Reply:
Done, that's better :)